### PR TITLE
[arrow] Update to 17.0.0

### PR DESCRIPTION
**Pull request by raulcd**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [x] The "supports" clause reflects platforms that may be fixed by this new version.
- [ ] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by MonicaLiu0311** [comment]
I got the following error when testing the `cuda` feature:
```
CMake Error at scripts/cmake/vcpkg_fixup_pkgconfig.cmake:134 (message):
  G:/arrow/downloads/tools/msys2/1e74ca60daa10104/mingw64/bin/pkg-config.exe
  --exists arrow-cuda failed with error code: 1

      ENV{PKG_CONFIG_PATH}: "G:/arrow/packages/arrow_x64-windows/lib/pkgconfig;G:/arrow/packages/arrow_x64-windows/share/pkgconfig;G:/arrow/installed/x64-windows/lib/pkgconfig;G:/arrow/installed/x64-windows/share/pkgconfig"
      output: Package cuda was not found in the pkg-config search path.

  Perhaps you should add the directory containing `cuda.pc'

  to the PKG_CONFIG_PATH environment variable

  Package 'cuda', required by 'arrow-cuda', not found
Call Stack (most recent call first):
  scripts/cmake/vcpkg_fixup_pkgconfig.cmake:196 (z_vcpkg_fixup_pkgconfig_check_files)
  ports/arrow/portfile.cmake:71 (vcpkg_fixup_pkgconfig)
  scripts/ports.cmake:192 (include)
```

**Comment by raulcd** [comment]
@kou any idea?

**Comment by raulcd** [comment]
ok, it seems the cuda dependency: https://github.com/microsoft/vcpkg/blob/492d1cc89fcf00c04bcfdca2e73fbbd3da97e849/ports/cuda/portfile.cmake#L1 doesn't really install cuda.
So cuda should be installed as a system dependency.
At this point, we might want to remove the `arrow-cuda` feature, thoughts?

File: ports/arrow/android.patch
Comment (kou) [inline ports/arrow/android.patch:1]: It seems that we can also remove this: https://youtube.com/live/gppMrYxoPeQ?feature=share

**Review by kou — COMMENTED** [review]
@MonicaLiu0311 Can CUDA be found with other port? I want to know that the CUDA not found error is only caused with this port.

FYI: Apache Arrow uses the standard `find_package(CUDAToolkit REQUIRED)`: https://github.com/apache/arrow/blob/apache-arrow-17.0.0/cpp/src/arrow/gpu/CMakeLists.txt#L44 
`FindCUDAToolkit.cmake` is included in CMake.

If it doesn't work with vcpkg, we may want to improve the cuda port.

**Comment by dg0yt** [comment]
@kou The error is from `vcpkg_fixup_pkgconfig` after the actual build. It points to the fact that arrow installs `arrow-cuda.pc` which refers to `cuda.pc` file even if such a file doesn't exist. That's why this error occurs only for `arrow`.

Obviously, `cuda.pc` should come with CUDA. Maybe CUDA does not provide pc files for Windows?

Possible fixes and workarounds.
- Make port `cuda` add a `cuda.pc` if it is missing.
- Remove `cuda` from `Requires[.private]` in `arrow-cuda.pc` (but ensure proper link `Libs[.private]`).
- Remove `arrow-cuda.pc` for Windows.

**Comment by kou** [comment]
Ah, I got it.

I downloaded the CUDA package for Windows https://developer.nvidia.com/cuda-downloads?target_os=Windows&target_arch=x86_64&target_version=11&target_type=exe_local and checked the content. It doesn't provide `cuda.pc`...

Can we keep the current configuration as-is for now? We will stop installing `arrow-cuda.pc` on Windows in upstream. So the `cuda` feature will work on Windows with Apache Arrow 18.0.0.

**Comment by MonicaLiu0311** [comment]
> @MonicaLiu0311 Can CUDA be found with other port? I want to know that the CUDA not found error is only caused with this port.

```
PS G:\vcpkg> ./vcpkg install pcl[cuda] --recurse
Computing installation plan...
The following packages are already installed:
    pcl[core,cuda]:x64-windows@1.14.1
pcl:x64-windows is already installed
Total install time: 1.74 ms
The package pcl provides CMake targets:

    find_package(PCL CONFIG REQUIRED)
    target_link_libraries(main PRIVATE ${PCL_LIBRARIES})
```

File: ports/arrow/android.patch
Comment (raulcd) [reply ports/arrow/android.patch:1]: @kou I think this is not the URL you wanted to share here :) so I am not sure what you wanted us to remove here

Comment (kou) [reply ports/arrow/android.patch:1]: Oh... Sorry... I've fixed the URL...
I wanted to show the latest code about this...

**Comment by kou** [comment]
Thanks.
It also shows that https://github.com/microsoft/vcpkg/pull/39948#issuecomment-2238046064 is right.
And https://github.com/microsoft/vcpkg/pull/39948#issuecomment-2238699172 is my suggestion to move forward this.

File: ports/arrow/android.patch
Comment (raulcd) [reply ports/arrow/android.patch:1]: I did update the patch to apply, it is still required because:
```diff
-if(NOT WIN32 AND NOT APPLE)
+if(NOT WIN32 AND NOT APPLE AND NOT ANDROID)
```
is necessary

Comment (kou) [reply ports/arrow/android.patch:1]: Ah, I see. I was wrong. Diff's diff is difficult to review...

Anyway, let's upstream and remove this patch in a future release: https://github.com/apache/arrow/issues/43390

**Comment by MonicaLiu0311** [comment]
Convert this PR to draft since there is no progress. Please ping us if this PR is ready for review again.

**Comment by raulcd** [comment]
> Convert this PR to draft since there is no progress. Please ping us if this PR is ready for review again.

@MonicaLiu0311 from my understanding @kou was asking you if we could keep the PR as is as this will get solved on 18.0.0, so I was just waiting for your feedback.

**Comment by raulcd** [comment]
ok, it seems a PR (with exactly the same changes as this original PR) was merged last week: https://github.com/microsoft/vcpkg/pull/40164/
I am closing this one now.

**Pull request closed without merging**